### ISSUE TITLE
Pie chart  fill target index returns negative values and breaks color indexing 

### ISSUE THREAD

**Issue by jimmykane** [amcharts/amcharts4#1783]
Hi, 

in most of my charts for filling up a series I use something like this on a pie series
```
    pieSeries.slices.template.adapter.add('fill', (fill, target, key) => {
      return this.getFillColor(target.parent.parent.chart, target.dataItem.index);
    });
```

where fillcolor is just: 

```
  getFillColor(chart: am4charts.XYChart | am4charts.PieChart, index: number) {
    return chart.colors.getIndex(index * 2);
  }
```

However on a pie chart the: 

```
target.dataItem.index
```

On the fill adapter, returns many times an index of -2 :-/ 

Here is a debug point inside the fill adapter call 

<img width="497" alt="Screenshot 2019-10-21 at 18 31 49" src="https://user-images.githubusercontent.com/1268723/67224390-5a798900-f431-11e9-9c28-86cb6ae130b7.png">

Is this a bug? If not what can I do to get cross chart consistent color filling when the data re the same ?(ordered and so on)

**Comment by zeroin** [comment]
I am looking into this. Do you mean -1? 
Actually I don't understand why you need this adapter at all, you do quite the same what the chart does itself.

**Comment by zeroin** [comment]
So it's actually a bug - we add slice to a parent and only then assign it a data item, so it returns -1 on a first adapter call. We will fix this issue in next release, meanwhile it should be safe to add simple if to check if the id is >=0.

**Comment by jimmykane** [comment]
@zeroin  that is great to hear. 

the reason I am overriding the fill color is that then I can control the 'step/index' multiplier via the base class call. 

So in this way the user can change that : 
```
 getFillColor(chart: am4charts.XYChart | am4charts.PieChart, index: number) {
    return chart.colors.getIndex(index * userSetting);
  }
```

And that applies a more consistent coloring 

Or do you think I am doing something wrong ?

**Comment by zeroin** [comment]
ColorSet has a `step` property: https://www.amcharts.com/docs/v4/reference/colorset/#step_property, so all you need to do is:

`chart.colors.step = 2;`

**Comment by jimmykane** [comment]
OMG. Ok loud and clear moving there

**Comment by jimmykane** [comment]
Ok but this does not work for all series. Still most of the series dont get a color assigned automatically 

For example column series

**Comment by martynasma** [comment]
Fixed in [4.7.4](https://github.com/amcharts/amcharts4/releases/tag/4.7.4).

## [4.7.4] - 2019-10-22

## Added
- New global option: `am4core.options.viewportTarget`. If you are placing charts into a scrollable container and using `onlyShowOnViewport`, set this option to a reference to the container. The system will monitor both window and individual scroll of the container to know where to reveal the lazy-loaded chart.
- New `MapChart` property: `geodataNames`. It can now be used to automatically translate country names into multiple languages. Check out [usage instructions](https://www.amcharts.com/docs/v4/tutorials/using-map-country-name-translations/).
- `hideParentColumns` property added to `TreeMap`. If set to `true` columns of the parent series will be hidden when user drills-down into deeper levels.

### Fixed
- `ColorSet` was starting to generate grayscale colors at some point.
- Accessibility: Series that has items with roles "menuitem" will now have its role set to "menu" automatically.
- Series bullets were dislocated on `XYChart` with `DateAxis` and `dateFormatter.utc = true`.
- In some specific cases 3D columns used incorrect `zIndex`.
- Index of dataItem of `PercentSeries` could be `-`1 after first render.
- `yRange` of `XYCursor` was inversed.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.

**Comment by martynasma** [comment]
Hey man, I'm closing this. If you would still like to pursue other questions that are not directly related to original topic, please open a new issue.
